gensim/models/doc2vec.py

```diff
@@ -808,6 +808,45 @@ def delete_temporary_training_data(self, keep_doctags_vectors=True, keep_inferen
         if self.docvecs and hasattr(self.docvecs, 'doctag_syn0_lockf'):
             del self.docvecs.doctag_syn0_lockf
 
+    def save_doc2vec_format(self, fname, word_vec=False, binary=False):
+        """
+        Store the input-hidden weight matrix.
+
+         `fname` is the file used to save the vectors in
+         `word_vec` is an optional boolean indicating whether to store word vectors
+         in the same file as document vectors
+         `binary` is an optional boolean indicating whether the data is to be saved
+         in binary word2vec format (default: False)
+
+        """
+        total_vec = len(self.docvecs)
+        if word_vec:
+            total_vec = total_vec + len(self.wv.vocab)
+        logger.info("storing %sx%s projection weights into %s" % (total_vec, self.vector_size, fname))
+
+        # save document vectors
+        with utils.smart_open(fname, 'wb') as fout:
+            fout.write(utils.to_utf8("%s %s\n" % (total_vec, self.vector_size)))
+            # store as in input order
+            for i in range(len(self.docvecs)):
+                doctag = self.docvecs.index_to_doctag(i)
```

Comment:
this will work in case user's `model.docvecs.doctags` is empty, and will assign the vector index as doctag